docs/typescript/statics.md

```diff
@@ -1,5 +1,37 @@
 # Statics in TypeScript
 
+To use Mongoose's automatic type inference to define types for your [statics](guide.html#statics) and [methods](guide.html#methods), you should define your methods and statics using the `methods` and `statics` schema options as follows.
+Do **not** use `Schema.prototype.method()` and `Schema.prototype.static()`.
+
+```typescript
+const userSchema = new mongoose.Schema(
+  { name: { type: String, required: true } },
+  {
+    methods: {
+      updateName(name: string) {
+        this.name = name;
+        return this.save();
+      }
+    },
+    statics: {
+      createWithName(name: string) {
+        return this.create({ name });
+      }
+    }
+  }
+);
+const UserModel = mongoose.model('User', userSchema);
+
+const doc = new UserModel({ name: 'test' });
+// Compiles correctly
+doc.updateName('foo');
+// Compiles correctly
+UserModel.createWithName('bar');
+```
+
+### With Generics
+
+While we recommend using Mongoose's automatic type inference where possible, you may need to override method and static types using generic parameters.
 Mongoose [models](../models.html) do **not** have an explicit generic parameter for [statics](guide.html#statics).
 If your model has statics, we recommend creating an interface that [extends](https://www.typescriptlang.org/docs/handbook/interfaces.html) Mongoose's `Model` interface as shown below.
```

Comment:
i think this sentence can be removed, because extending `Model` or `Document` is discouraged to my knowledge (and the example also does not do it)

Reply:
I updated this sentence a bit. There's nothing wrong with extending Model, it is often necessary if you're using generics.